src/RemoteHttpInterceptor.ts

```diff
@@ -67,45 +68,52 @@ export class RemoteHttpInterceptor extends BatchInterceptor<
       process.send?.(`request:${serializedRequest}`)
 
       const responsePromise = new Promise<void>((resolve) => {
-        handleParentMessage = (message) => {
-          if (typeof message !== 'string') {
-            return resolve()
-          }
-
-          if (message.startsWith(`response:${requestId}`)) {
-            const [, serializedResponse] =
-              message.match(/^response:.+?:(.+)$/) || []
+        this.requestId2Response.set(requestId, { request, resolve })
+      })
+      return responsePromise
+    })
 
-            if (!serializedResponse) {
-              return resolve()
-            }
+    const handleParentMessage: NodeJS.MessageListener = (message) => {
+      if (typeof message !== 'string') {
+        return
+      }
+      const startIdPosition = 'response:'.length;
+      const endIdPosition = message.indexOf(':', startIdPosition);
+      const requestId = message.slice(startIdPosition, endIdPosition)
```

Comment:
Instead of having it serialised manually I recommend putting the structure straight into the IPC message, since Node.js does JSON internally by itself. I'd do something like: `{ type: 'request', payload: {...}`.

Would you be open for such an enhancement?